lib/src/main/java/com/dynatrace/metric/util/DynatraceMetadataEnricher.java

```diff
@@ -0,0 +1,189 @@
+      throw new IOException("passed Reader cannot be null.");
+    }
+
+    String metadataFileName = null;
+
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
+      String line = reader.readLine();
+      if (line != null && !line.isEmpty()) {
+        metadataFileName = line;
+      }
+    }
+    return metadataFileName;
+  }
+
+  /**
+   * Read the actual content of the metadata file.
+   *
+   * @param fileContents A {@link Reader} object containing the metadata contents.
+   * @return A {@link List<String>} containing the {@link String#trim() trimmed} lines.
+   * @throws IOException if an error occurs during reading of the file.
+   */
+  static List<String> getDynatraceMetadataFileContents(Reader fileContents) throws IOException {
+    if (fileContents == null) {
+      throw new IOException("passed Reader cannot be null.");
+    }
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
+      return reader.lines().map(String::trim).collect(Collectors.toList());
+    }
+  }
+
+  /**
+   * A helper function that returns whether file exists and is readable.
+   *
+   * @param filePath The path to the file.
+   * @return true if the file exists and is readable and false otherwise.
+   */
+  static boolean fileExistsAndIsReadable(String filePath) {
+    try {
+      File file = new File(filePath);
+      if (file.exists() && Files.isReadable(file.toPath())) {
+        return true;
+      }
+    } catch (Exception ignored) {
+      // something went wrong, but at this point we don't care what it was.
+    }
+    return false;
+  }
+
+  /**
+   * Gets the file location of the metadata file from the indirection file and reads the contents
+   * thereof. If the indirection file does not exist, falls back to the alternative metadata file.
+   *
+   * @return A {@link List<String>} representing the contents of the Dynatrace metadata file.
+   *     Leading and trailing whitespaces are {@link String#trim() trimmed} for each of the lines.
+   */
+  static List<String> getMetadataFileContentWithRedirection(
+      String indirectionFileName, String alternativeMetadataFilename) {
+    String metadataFileName = null;
+
+    try (Reader indirectionFileReader = new FileReader(indirectionFileName)) {
+      metadataFileName = getMetadataFileName(indirectionFileReader);
+    } catch (FileNotFoundException e) {
+      logger.info("Indirection file not found. This is normal if OneAgent is not installed.");
+    } catch (Exception e) {
+      logger.info(
+          String.format("Error while trying to read contents of OneAgent indirection file: %s", e));
+    }
+
+    if (metadataFileName == null || metadataFileName.isEmpty()) {
+      if (DynatraceMetadataEnricher.fileExistsAndIsReadable(alternativeMetadataFilename)) {
+        // alternative file exists, use it for metadata enrichment
+        logger.info(
+            String.format(
+                "Alternate metadata file exists, attempting to read from %s.",
```

Comment:
```suggestion
                "Alternative metadata file exists, attempting to read from %s.",
```
Would be more common I think.